integration_tests/tests/transactionCSV-to-s3-event-tests.ts

```diff
@@ -1,97 +1,70 @@
-      ]),
-    });
-    // Upload the csv file to S3 transaction csv bucket
-    const testObject: S3Object = {
-      bucket: `${prefix}-transaction-csv`,
-      key: `fakeBillingReport.csv`,
-    };
-    await putS3Object({ target: testObject, data: Buffer.from(csvString) });
-  });
+    // Add data to build the test csv
 
-  it("stores valid events in the storage/btm_transactions/yyyy-mm-dd folder", async () => {
-    const checkEventsExistsInS3 = await waitForTrue(
-      checkS3BucketForEventIds,
-      1000,
-      10000
+    // Upload the csv file to S3 transaction csv bucket
+    await putS3Object({
+      target: {
+        bucket: inputBucket,
+        key: `fakeBillingReport.csv`,
+      },
+      data: csv,
+    });
+    await poll<ListObjectsCommandOutput>(
+      async () =>
+        await listS3Objects({
+          bucketName: outputBucket,
+          prefix: transactionsDirectory,
+        }),
+      (result) =>
+        result.Contents?.length !== undefined && result.Contents?.length > 0
     );
-    expect(checkEventsExistsInS3).toBe(true);
-    for (let i = 0; i < testCases.length; i++) {
-      const s3Object = await getS3Object({
-        bucket: bucketName,
-        key: `${folderPrefix}/${"2023-01-01"}/${
-          testCases[i].expectedEventId
-        }.json`,
-      });
-
-      if (testCases[i].expectedPath === "happy" && s3Object !== undefined) {
-        const s3Event = JSON.parse(s3Object);
-        expect(s3Event.client_id).toEqual(testCases[i].expectedClientId);
-        expect(s3Event.event_id).toEqual(testCases[i].expectedEventId);
-        expect(s3Event.event_name).toEqual(testCases[i].expectedEventName);
-      }
-    }
   });
 
-  it("does not store invalid events in the storage/btm-transactions/yyyy-mm-dd folder", async () => {
-    for (let i = 0; i < testCases.length; i++) {
+  it("stores stores the events we care about in the storage bucket", async () => {
+    for (const testCase of testCases) {
       const s3Object = await getS3Object({
-        bucket: bucketName,
-        key: `${folderPrefix}/${"2023-01-01"}/${
-          testCases[i].expectedEventId
+        bucket: outputBucket,
+        key: `${transactionsDirectory}/${"2023-01-01"}/${
+          testCase.expectedEventId
         }.json`,
       });
-      if (testCases[i].expectedPath === "sad") {
-        expect(s3Object).toEqual("NoSuchKey");
+      switch (testCase.expectedPath) {
+        case "happy": {
+          if (s3Object === undefined)
+            throw new Error("No event was made for a happy path");
+          const s3Event = JSON.parse(s3Object);
+          expect(s3Event.client_id).toEqual(testCase.expectedClientId);
+          expect(s3Event.event_id).toEqual(testCase.expectedEventId);
+          expect(s3Event.event_name).toEqual(testCase.expectedEventName);
+          break;
+        }
+        case "sad": {
+          expect(s3Object).toEqual("NoSuchKey");
+          break;
+        }
```

Comment:
I've bundled both test segments into one loop through the `testCases` array and now we just switch what we're `expect`ing based on the path